doc/bucket-types.md

```diff
@@ -0,0 +1,41 @@
+[NooBaa Operator](../README.md) /
+
+# Bucket Types in NooBaa
+NooBaa supports two types of buckets - most commonly referred to as `namespace` and `data`. While all NooBaa buckets are accessible via the S3 API and appear the same to the user, they have different uses and implementations under the hood.
+This document will explaing the meaning of each, as well as the differences between the two.
+
+## Data Buckets
+Data buckets are the 'classic' type of bucket in NooBaa. In a classic NooBaa deployment, the product creates several resources by default - a backingstore (`noobaa-default-backing-store`), a bucketclass (`noobaa-default-bucket-class`) and a bucket (`first.bucket`) as part of the product's initialization process. When data is written to these buckets, it is first processed by NooBaa - the data is compressed, deduplicated, encrypted, and split into chunks. These chunks are then stored in the backingstore that the bucket is connected to.
+The only way to access the objects on data buckets is through NooBaa - the chunks cannot be used or deciphered without the system.
+
+## Namespace Buckets
+Namespace buckets work differently than data buckets, since they try¹ to not apply any processing on the objects that are uploaded to them, and act as more of a 'passthrough'. In cases where a non-S3 storage provider API is used, NooBaa takes care of bridging potential gaps between S3 and the provider's API - no user action is required. This also means that as long as NooBaa supports a certain provider's API, users can use the S3 API to manage it (while NooBaa takes care of the 'translation')
+The objects can be accessed both from within NooBaa, as well as from whatever storage providers hosts them.
+Namespace buckets also support having multiple 'read' sources - for example, a single namespace bucket can show the objects from several Azure Blob containers, AWS S3 buckets, Google Cloud Platform buckets, all at the same time.
+Another feature of namespace buckets is cache, which can be configured to keep frequently-accessed objects in the system's memory, effectively reducing access time as well as egress and access costs.
+
+_1. In some cases, object metadata (such as tags) might have to be modified in order to comply with a cloud provider's limits_
+
+## Supported Storage Providers by Bucket Type
+| Service                                     | Data Buckets | Namespace Buckets |
+|---------------------------------------------|     :---:    |        :---:      |
+| Amazon Web Services S3 (incl. STS support)  | ✅          | ✅
+| S3 Compatible Services                      | ✅          | ✅
+| Azure Blob Storage                          | ✅          | ✅
+| Google Cloud Platform Object Storage        | ✅          | ❌
+| Filesystem (NSFS)                           | ❌          | ✅
```

Comment:
I would add to the rest the brackets with the abbreviation/acronym - AWS, GCP, PV, etc.